query-algebrizer/src/clauses/or.rs

```diff
@@ -407,20 +407,24 @@ impl ConjoiningClauses {
             // ```
             let mut receptacles =
                 patterns.into_iter()
-                        .filter_map(|pattern| {
+                        .map(|pattern| {
                             let mut receptacle = template.make_receptacle();
                             println!("Applying pattern with attribute {:?}", pattern.attribute);
                             receptacle.apply_pattern_clause_for_alias(schema, &pattern, &source_alias);
-                            if receptacle.is_known_empty() {
-                                println!("Receptacle is empty.");
-                                let reason = receptacle.empty_because;
-                                None
-                            } else {
-                                Some(receptacle)
-                            }
+                            receptacle
                         })
                         .peekable();
 
+            // Let's see if we can grab a reason if every pattern failed.
```

Comment:
I don't think this expression is wrong, but it's really pretty awkward.  Consider just building a `Vec` to avoid `peekable`.